marlowe-protocols/src/Network/Protocol/Connection.hs

```diff
@@ -1,200 +1,50 @@
-{-# LANGUAGE DataKinds #-}
```

Comment:
Major refactoring of connection. The new version is much simpler and doesn't assume the use of binary channels.